b2g/chrome/content/remote_command.js

```diff
@@ -0,0 +1,492 @@
+function handleClickEvent(event)
+{
+  let isCursorMode = getIsCursorMode();
+
+  if(isCursorMode) {
+    let type = 'navigator:browser';
+    let shell = Services.wm.getMostRecentWindow(type);
+    let utils = shell.QueryInterface(Components.interfaces.nsIInterfaceRequestor)
+                     .getInterface(Components.interfaces.nsIDOMWindowUtils);
+    let cursorPos = getCursorPosition();
+
+    ["mousedown",  "mouseup"].forEach(function(mouseType) {
+      utils.sendMouseEvent(mouseType, cursorPos["x"], cursorPos["y"], 0, 1, 0);
+     });
+
+     if (event.action == "dblclick") {
+       ["mousedown",  "mouseup"].forEach(function(mouseType) {
+         utils.sendMouseEvent(mouseType, cursorPos["x"], cursorPos["y"], 0, 2, 0);
+       });
+     }
+  } else {
+    // Send enter key when not in cursor mode
+    handleKeyboardEvent('DOM_VK_RETURN');
+  }
+}
+
+// sendMouseEvent in cursor mode to move cursor position, or dispatch 4 arrow keys from swipe
+// Receved event format:
+// { type: 'command', action: 'touchstart', detail: { width: <client panel width>, height: <client panel height>} }
+// { type: 'command', action: 'touchmove', detail: { dx: <dx to start point>, dy: <dy to start point>} }
+// { type: 'command', action: 'touchend', detail: { dx: <dx to start point>, dy: <dy to start point>}, swipe: <gesture of the touch, up/down/left/right> }
+function handleTouchEvent(event)
+{
+  let type = 'navigator:browser';
+  let shell = Services.wm.getMostRecentWindow(type);
+  let utils = shell.QueryInterface(Components.interfaces.nsIInterfaceRequestor)
+                   .getInterface(Components.interfaces.nsIDOMWindowUtils);
+  let cursorPos = getCursorPosition();
+  let x = cursorPos["x"];
+  let y = cursorPos["y"];
+  let startX = 0;
+  let startY = 0;
+  let isCursorMode = getIsCursorMode();
+
+  // Each event is independent, when touchstart, we set startX/startY for following touchmove/touchend
+  switch (event.action) {
+    case "touchstart":
+      startX = cursorPos["x"];
+      startY = cursorPos["y"];
+      setSharedState("startX", startX.toString());
+      setSharedState("startY", startY.toString());
+      break;
+    case "touchmove":
+    case "touchend":
+      startX = parseInt(getSharedState("startX"));
+      startY = parseInt(getSharedState("startY"));
+      break;
+    default:
+      return;
+  }
+
+  // Calcualte new cursor position based on startX/dx, startY/dy when receiving each touchmove/touchend
+  // Limit x/y greater than 0 and less than innerWidth/innerHeight
+  let detail = event.detail;
+  x = Math.min(shell.innerWidth, Math.max(0, startX + detail.dx));
+  y = Math.min(shell.innerHeight, Math.max(0, startY + detail.dy));
+
+  setSharedState("x", x.toString());
+  setSharedState("y", y.toString());
+
+  if(isCursorMode) {
+    utils.sendMouseEvent("mousemove", x, y, 0, 0, 0);
+
+    // Use SystemAppProxy send chrome event to control gaia mouse cursor
+    sendChromeEvent('move-cursor', {
```

Comment:
Are we still using Gaia-rendered cursor? If not, please remove the related code.